hack/update-bazel-with-image.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+if [ -x "$(command -v docker)" ]; then
+    CONTAINER_ENGINE=docker
+elif [ -x "$(command -v podman)" ]; then
+    CONTAINER_ENGINE=podman
+else
+	echo "There is no docker or podman installed. Please use hack/update-bazel.sh script instead."
+	exit 1
+fi
+
+WORKDIR=$(pwd)
+TEST_INFRA_PATH=${WORKDIR%/hack}
+
+
+$CONTAINER_ENGINE run --volume ${TEST_INFRA_PATH}:/test-infra:Z \
+--workdir /test-infra \
+-it gcr.io/k8s-testimages/launcher.gcr.io/google/bazel:latest-test-infra \
```

Comment:
Can you also fix perms to use the userid so we don't end up creating files owned by root?

Reply:
well, in theory, the developer could work as root and commit his changes with these perms. I don't think that this script will affect anything. That depends on the developer.

Reply:
no it does not depend on the developer. This script needs to include `--user=$UID` so it will run as the same UID as the user that is calling it

Reply:
So the best practice here would be `--user $(id -u):$(id -g)` which for some reason doesn't work on me. I guess the problem is on my side. I would appreciate it if you check it too...

Reply:
works fine:
```
 docker run --rm  --user $(id -u):$(id -g)  debian id
uid=1000 gid=1000 groups=1000
```